pkg/cluster/adminupdate_test.go

```diff
@@ -130,7 +130,9 @@ func TestAdminUpdateSteps(t *testing.T) {
 				"[Action fixMCSCert-fm]",
 				"[Action fixMCSUserData-fm]",
 				"[Action ensureGatewayUpgrade-fm]",
+				"[Condition aroDeploymentReady-fm, timeout 5m0s]",
 				"[Action rotateACRTokenPassword-fm]",
+				"[Condition ensureValidACRToken-fm, timeout 5m0s]",
```

Comment:
Do you have thoughts on consolidating these? i.e, convert rotateACRTokenPassword-fm to a Condition with timeout, have your validate path be triggered as part of `rotateACRTokenPassword-fm` instead of a separate step?

Reply:
That makes sense, I'll combine it all into a single condition.